node/protocols/data_query/data_query.go

```diff
@@ -43,6 +43,7 @@ type Interface interface {
 	GetQueryResponse(key string) ([]messages.DataQueryResponse, bool)
 	SendDataQueryResponse(ctx context.Context, peerID peer.ID, payload *messages.DataQueryResponseProto) error
 	RequestDataQueryResponseTransfer(ctx context.Context, peerID peer.ID, request *messages.DataQueryResponseTransferProto) error
+	PurgeQueryHistory(key string) error
```

Comment:
Thanks for the PR, some comments from my side. We should convert `PurgeQueryHistory` to `PurgeQueryHistory() error` and go through all the data query requests and if they are older than 5 mins delete them.

Reply:
@filefilegoadmin done